src/charm.py

```diff
@@ -132,6 +134,7 @@ def __init__(self, *args):
 
         # register charm actions
         actions = [
+            self.on.restart_action,
```

Comment:
This actually REGISTERS the restart action -- it wasn't ever handled by the charm code !?!